src/setup/node.rs

```diff
@@ -90,17 +96,32 @@ impl NodeBuilder {
             fs::create_dir_all(&target)?;
         }
 
+        let setup_path = build_ripple_work_path()?.join(RIPPLE_SETUP_DIR);
+
         match node_type {
             NodeType::Stateful => {
-                let source = get_stateful_node_path()?;
+                let node_idx = self
+                    .stateful_nodes_pool
+                    .pop()
+                    .expect("Not enough available stateful nodes");
```

Comment:
Perhaps `Not enough stateful nodes available` sounds better?

Reply:
As this function returns `Result` maybe we shouldn't do `expect` here?

Reply:
I agree.

Reply:
> As this function returns Result maybe we shouldn't do expect here?

Not anymore. But I applied your first suggestion since it's still applicable.

Done